### PR TITLE
v1.3: `attributesToSearchOn`

### LINKED ISSUE #2478: v1.3: Define search field at search time
v1.3 introduces a new search parameter, `attributesToSearchOn`.

`attributesToSearchOn` is a search parameter accepting an array of strings indicating one or more document attributes. Queries using `attributesToSearchOn` will restrict the search to the indicated attributes.

Given the following dataset:

```json
{
  "id": 0,
  "name": "Our Wives Under the Sea",
  "genre": ["horror", "scifi"],
  "synopsis": "A woman returns to her wife transformed after a deep-sea adventure."
},
{
  "id": 1,
  "name": "A Strange and Stubborn Endurance",
  "genre": ["adventure"],
  "synopsis": "A man must overcome trauma and fight off a mysterious assassin."
}
```

And the following query:

```json
{
  "q": "adventure",
  "attributesToSearchOn": ["genre"]
}
```

Meilisearch will only return document `1`.

Both documents contain the word adventure, but `"attributesToSearchOn": ["genre"]` instructs Meilisearch to only consider results found on the `genre` field.

## Tasks
- [x] Add `attributesToSearchOn` to `/reference/api/search#search_parameters`
- [x] Update search parameter tables in `/reference/api/search` and `/reference/api/multi_search`
- [ ] (Optional) Add `<Capsule />` mentioning `attributesToSearchOn` to `/learn/fine_tuning_results/filtering`

## References
- engine issue: https://github.com/meilisearch/meilisearch/issues/3772
- engine PR: https://github.com/meilisearch/meilisearch/pull/3834
- PRD (private): https://www.notion.so/meilisearch/Define-Fields-To-Search-On-At-Search-Time-0725a20ed5bd46b593683a67697d3e7b
- SME: @ManyTheFish

### PR DESCRIPTION
**Pull request by guimachiavelli**
Closes #2478

### CODE REVIEW

File: reference/errors/error_codes.mdx [hunk 1]
```diff
@@ -126,6 +126,10 @@ The [`offset`](/reference/api/keys#query-parameters) parameter is invalid. It sh
 
 The given [`uid`](/reference/api/keys#uid) is invalid. The `uid` must follow the [uuid v4](https://www.sohamkamani.com/uuid-versions-explained) format.
 
+## `invalid_attributes_to_search_on`
```
Comment (ManyTheFish) [inline reference/errors/error_codes.mdx:129]:
```suggestion
## `invalid_search_attributes_to_search_on`
```

related to https://github.com/meilisearch/documentation/issues/2478#issuecomment-1617782149

Comment (guimachiavelli) [reply reference/errors/error_codes.mdx:129]: Brilliant, thanks!

**Review by brunoocasali — APPROVED** [review]
The code-samples are ok! 👍

**Comment by bidoubiwa** [comment]
This might be updated if these PR's are merged: 

https://github.com/meilisearch/meilisearch/issues/3912
https://github.com/meilisearch/meilisearch/issues/3911

**Review by maryamsulemani97 — APPROVED** [review]
🦔

**Review by ManyTheFish — APPROVED** [review]

**Pull request merged**